core/src/main/java/org/apache/iceberg/io/StructCopy.java

```diff
@@ -21,8 +21,8 @@
 import org.apache.iceberg.StructLike;
 
 /** Copy the StructLike's values into a new one. It does not handle list or map values now. */
-class StructCopy implements StructLike {
```

Comment:
I need to move this into a utility. Just not sure which at this point. Ideas are welcome.

Reply:
Does this even belong in `io`?  This feels like it should be in `types.TypeUtil`

Reply:
I'd say it belongs to `StructLikeUtil` or something like this. Doesn't seem to belong to `io`.

Reply:
+1 for `StructLikeUtil` or similar.

@jbonofre, the `io` classes typically refer to `FileIO` and buffer management utils in Iceberg. I'm not quite sure why this is currently in the `io` package. I think it's probably better to not continue using `io` so that it's easier to find later.

Reply:
Good point. I like the `StructLikeUtil` name.

+1 for that. Thanks !

Reply:
Moved the logic to the util and deprecated this class not to break anyone.